src/main/java/uk/gov/companieshouse/web/pps/validation/EnterDetailsValidator.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.companieshouse.web.pps.validation;
+
+import static java.util.Locale.UK;
+import static java.util.ResourceBundle.getBundle;
+
+import java.util.ResourceBundle;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import uk.gov.companieshouse.web.pps.models.EnterDetails;
+import uk.gov.companieshouse.web.pps.util.PenaltyReference;
+
+@Component
+public class EnterDetailsValidator {
+
+    private static final String LATE_FILING_PENALTY_REF_REGEX = "^A\\d{7}$";
+    private static final String SANCTIONS_PENALTY_REF_REGEX = "^PN\\d{8}$";
+
+    private final ResourceBundle bundle;
+
+    public EnterDetailsValidator() {
+        this.bundle = getBundle("ValidationMessages", UK);
+    }
+
+    public void isValid(final EnterDetails enterDetails, final BindingResult bindingResult) {
+        String penaltyRef = enterDetails.getPenaltyRef();
+        String penaltyRefErrorText = switch (PenaltyReference.valueOf(enterDetails.getPenaltyReferenceName())) {
+            case LATE_FILING -> getPenaltyRefErrorText(penaltyRef, "enterDetails.penaltyRef.isEmpty.LATE_FILING",
+                    LATE_FILING_PENALTY_REF_REGEX, "enterDetails.penaltyRef.isWrong.LATE_FILING");
+            case SANCTIONS -> getPenaltyRefErrorText(penaltyRef, "enterDetails.penaltyRef.isEmpty.SANCTIONS",
+                    SANCTIONS_PENALTY_REF_REGEX, "enterDetails.penaltyRef.isWrong.SANCTIONS");
+        };
```

Comment:
I am a bit concerned that once we start adding more penalty types that this switch statement will end up quite large. What about using a MultiMap and instantiating the various values and then drive this from the method call? I used a a few maps just to test out the approach, but I think apache commons or guava multimap would be a good candidate. Ex:
`public void isValid(final EnterDetails enterDetails, final BindingResult bindingResult) {
        String penaltyRef = enterDetails.getPenaltyRef();
        PenaltyReference penaltyReferenceName = PenaltyReference.valueOf(enterDetails.getPenaltyReferenceName());
        Optional<String> penaltyRefErrorText = getPenaltyRefErrorText(penaltyRef, penaltyReferenceName);
        penaltyRefErrorText.ifPresent(
                s -> bindingResult.rejectValue("penaltyRef", "penaltyRef", s));
    }

    private Optional<String> getPenaltyRefErrorText(String penaltyRef, PenaltyReference penaltyReference) {
        final String penaltyRefIsEmptyErrorMessageKey = emptyMsgKeyMap.get(penaltyReference);
        final String penaltyRefIsWrongErrorMessageKey = incorrectMsgEx.get(penaltyReference);
        final String penaltyRefRegex = regExMap.get(penaltyReference);

        Optional<String> penaltyRefErrorText = Optional.empty();
        if (penaltyRef.isEmpty()) {
            penaltyRefErrorText = Optional.of(bundle.getString(penaltyRefIsEmptyErrorMessageKey));
        } else if (!penaltyRef.matches(penaltyRefRegex)) {
            penaltyRefErrorText = Optional.of(bundle.getString(penaltyRefIsWrongErrorMessageKey));
        }

        return penaltyRefErrorText;
    }`

Reply:
Don't think there is much need to be concerned here @acallaghan-ch. Looking at the Epics in the backlog we only have another journey to introduce [SAN-13](https://companieshouse.atlassian.net/browse/SAN-13) ROE UI. That is also reflected in the prototype, see [Penalty details (Register an overseas entity)](https://sanctions-payment-prototype-0d6d68d6c82d.herokuapp.com/v9/04c-penalty-details-roe).  Therefore I'm not planning on changing this any further in this ticket / PR.

Reply:
I think they said that eventually there will be up to 200 different penalty types, so it was really with one eye on the future with this.

Reply:
Depends whether we expect many more penalty types or not here, as I understood the SANCTIONS would cover a large amount of types, and we might need to add ROE but nothing else.